From 955745bc12fc8d873b804a11f6ab154ca1cab3dc Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Thu, 1 Mar 2018 14:24:33 +0800
Subject: [PATCH] drm/bridge: synopsys: dw-hdmi: disable phy in dw_hdmi_bind

If hdmi is enabled in uboot and pluged out when booting kernel,
the hdmi phy is still enabled. It's better to disable it to
match the real status.

Change-Id: Ia1c5ede6499ee277d08c35a85c50e3257305f90f
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
index 64f854f2c549..5ab3d816cc6e 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi.c
@@ -3614,6 +3614,8 @@ int dw_hdmi_bind(struct device *dev, struct device *master,
 		hdmi->disabled = false;
 		hdmi->bridge_is_on = true;
 		hdmi->phy.enabled = true;
+	} else if (ret & HDMI_PHY_TX_PHY_LOCK) {
+		hdmi->phy.ops->disable(hdmi, hdmi->phy.data);
 	}
 	init_hpd_work(hdmi);
 	initialize_hdmi_ih_mutes(hdmi);
-- 
2.35.3

